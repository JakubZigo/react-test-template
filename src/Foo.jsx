import React, { Component } from 'react'

class Foo extends Component {
  constructor(props) {
    super(props)
    this.state = {
      data: 'Initial data'
    }
  }

  updateState() {
    this.setState({data: 'Data updated'})
  }

  render() {
    return (
      <div className="foo">
        <button onClick={() => this.updateState()}>CLICK</button>
        <h4>{this.state.data}</h4>
      </div>
    )
  }
}

export default Foo
