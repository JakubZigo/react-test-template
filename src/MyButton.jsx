import React, { Component } from 'react'

export default class MyButton extends Component {
  constructor(props) {
    super(props)
    this.state = {
      clickCount: 0
    }
  }

  handleClick(text) {
    this.props.clickFunction(text)
    this.setState({
      clickCount: this.state.clickCount + 1
    })
  }

  render() {
    return (
      <button id="button" onClick={() => this.handleClick('hello world')}>button</button>
    )
  }
}

