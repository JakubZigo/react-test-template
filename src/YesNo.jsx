import React, { Component } from 'react'
import { connect } from 'react-redux'
import { getAnswer } from '../actions/yesno'

export class YesNo extends Component {
  constructor(props) {
    super(props)
  }

  render() {
    return (
      <div>
        <button id="getAnswerButton" onClick={() => this.props.getAnswer()}>YES | NO</button>
        <h4>{this.props.answer}</h4>
        <img src={this.props.image} />
      </div>
    )
  }
}

/* istanbul ignore next */
const mapStateToProps = (state) => {
  console.log(state)
  return {
    answer: state.yesno.answer,
    image: state.yesno.image,
  }
}

/* istanbul ignore next */
const mapDispatchToProps = (dispatch) => {
  return {
    getAnswer: () => dispatch(getAnswer()),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(YesNo)
