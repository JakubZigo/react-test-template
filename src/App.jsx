import React, { Component } from 'react'
import Foo from './Foo.jsx'
import YesNo from './YesNo.jsx'

class App extends Component {
  render() {
    return (
      <div>
        <Foo />
        <YesNo />
      </div>
    )
  }
}

export default App

