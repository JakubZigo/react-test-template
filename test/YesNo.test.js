import React from 'react'
import { shallow, mount, render } from 'enzyme'
import toJson from 'enzyme-to-json'
import sinon from 'sinon'

import { YesNo } from '../src/YesNo'

describe('YesNo', () => {
  let getAnswerSpy

  beforeEach(() => {
    getAnswerSpy = sinon.spy()
  })

  it('should render h4 with answer and img', () => {
    const wrapper = shallow(<YesNo answer='answer' image='image_url'/>)
    expect(wrapper.find('h4').length).toBe(1)
    expect(wrapper.find('h4').at(0).text()).toBe('answer')
    expect(wrapper.find('img').length).toBe(1)
    expect(wrapper.find('img').at(0).props()).toEqual({src: 'image_url'})
  })

  it('should request answer when getAnswer button is clicked', () => {
    const wrapper = shallow(<YesNo getAnswer={() => getAnswerSpy()}/>)
    wrapper.find('#getAnswerButton').simulate('click')
    expect(getAnswerSpy.called)
  })

  it('should render correctly', () => {
    const wrapper = shallow(<YesNo />)
    expect(wrapper).toMatchSnapshot();
  })
})
