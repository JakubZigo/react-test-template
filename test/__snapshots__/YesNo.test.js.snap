// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YesNo should render correctly 1`] = `
<div>
  <button
    id="getAnswerButton"
    onClick={[Function]}
  >
    YES | NO
  </button>
  <h4 />
  <img />
</div>
`;
